modules/base/rendering/renderableswitch.cpp

```diff
@@ -0,0 +1,141 @@
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/io/texture/texturereader.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/crc32.h>
+#include <ghoul/misc/profiling.h>
+#include <ghoul/opengl/texture.h>
+#include <fstream>
+#include <optional>
+
+ //renderableSelectRenderable
+ //add by distance to the end
+ //renderableSelectorByDistance
+ //RenderableSwitcherByDistance
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo AutoScaleInfo = {
+        "AutoScale",
+        "Auto Scale",
+        "When true, the plane will automatically adjust in size to match the aspect "
+        "ratio of the content. Otherwise it will remain in the given size."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo DistanceThresholdInfo = {
+        "DistanceThreshold",
+        "Distance Threshold",
+        "Threshold for when the switch happens between the two renderables. "
+    };
+
+    struct [[codegen::Dictionary(RenderableSwitch)]] Parameters {
+        ghoul::Dictionary Renderable1;
+        ghoul::Dictionary Renderable2;
+
+        std::optional<float> distanceThreshold;
+    };
+#include "renderableswitch_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+    documentation::Documentation RenderableSwitch::Documentation() {
+        return codegen::doc<Parameters>(
+            "base_renderable_switch"
+        );
+    }
+
+    RenderableSwitch::RenderableSwitch(const ghoul::Dictionary& dictionary)
+        : Renderable(dictionary), _autoScale(AutoScaleInfo, false)
+        , _distanceThreshold(DistanceThresholdInfo, 1000000.f)
+    {
+        const Parameters p = codegen::bake<Parameters>(dictionary);
+
+        _renderable1 = createFromDictionary(p.Renderable1);
+
+        _renderable2 = createFromDictionary(p.Renderable2);
+
+        _distanceThreshold = p.distanceThreshold.value_or(_distanceThreshold);
+        addProperty(_distanceThreshold);
+
+        addProperty(_autoScale);
+    }
+
+    void RenderableSwitch::initializeGL() {
+        if (_renderable1) {
+            _renderable1->initializeGL();
+        }
+        if (_renderable2) {
+            _renderable2->initializeGL();
+        }
+    }
+
+    void RenderableSwitch::deinitializeGL() {
+        if (_renderable1) {
+            _renderable1->deinitializeGL();
+        }
+        if (_renderable2) {
+            _renderable2->deinitializeGL();
+        }
+    }
+
+    bool RenderableSwitch::isReady() const {
+        return (_renderable1 && _renderable1->isReady()) &&
+            (_renderable2 && _renderable2->isReady());
+    }
+
+    void RenderableSwitch::update(const UpdateData& data) {
+        if (_renderable1) {
+            _renderable1->update(data);
+        }
+        if (_renderable2) {
+            _renderable2->update(data);
+        }
+    }
+
+    void RenderableSwitch::render(const RenderData& data, RendererTasks& tasks) {
+        if (!_enabled) {
+            return;
+        }
```

Comment:
This shouldn't be necessary as if the renderable is disabled, the `render` function will not be called